src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -3136,21 +3136,15 @@ private bool TryProcessEnvChange(int tokenLength, TdsParserStateObject stateObj,
                             _defaultCollation = env._newCollation;
                             _defaultLCID = env._newCollation.LCID;
 
-                            int newCodePage = GetCodePage(env._newCollation, stateObj);
 
                             // UTF8 collation
                             if (env._newCollation.IsUTF8)
                             {
                                 _defaultEncoding = Encoding.UTF8;
-
-                                if (newCodePage != _defaultCodePage)
-                                {
-                                    _defaultCodePage = newCodePage;
-                                }
                             }
                             else
                             {
-
+                                int newCodePage = GetCodePage(env._newCollation, stateObj);
```

Comment:
Do we need to make the same change in the netcore file?

Reply:
netcore already has it.